src/lib.rs

```diff
@@ -69,11 +69,34 @@ impl Order {
         }
     }
 
+    /// Generate a Certificate Signing Request for the given names and request finalization
```

Comment:
One refactor I was hoping to figure out is to have the `Order` know the `Identifier`'s it was created with. After that I think the desired API shape for this operations would be something like `let cert_pem = my_order.finalize().await?` without any args. 

Asking the user to provide the `Identifiers` at the time of finalization like this API will be prone to error. [RFC 8555 7.4](https://www.rfc-editor.org/rfc/rfc8555#section-7.4) says the server has to return an error if the CSR's names and the order's identifiers aren't an exact match.

Reply:
Fixed this!